### PR TITLE
Introduce SymbolObjectCodeRequest

### PR DESCRIPTION
**Pull request by hamishknight**
Add `SymbolObjectCodeRequest`, which emits object code for a specific set of symbols by querying TBDGen for the source of the symbols, and asking SILGen (and eventually IRGen) to emit them.

### CODE REVIEW

**Comment by hamishknight** [comment]
@swift-ci please test

File: lib/IRGen/IRGen.cpp [hunk 1]
```diff
@@ -79,6 +81,7 @@
 #include "llvm/Transforms/Instrumentation/SanitizerCoverage.h"
 #include "llvm/Transforms/Instrumentation/ThreadSanitizer.h"
 #include "llvm/Transforms/ObjCARC.h"
+#include "llvm/Transforms/Utils/Cloning.h"
```
Comment (CodaFi) [inline lib/IRGen/IRGen.cpp:84]: Curious what motivates this header dependency.

**Review by CodaFi — APPROVED** [review]

File: lib/IRGen/IRGen.cpp (hunk 1 above)
Comment (hamishknight) [reply lib/IRGen/IRGen.cpp]: Ah that's a holdover from https://github.com/apple/swift/pull/32964 when we cloned the module with the requested symbols – good spot!

**Comment by hamishknight** [comment]
@swift-ci please smoke test

**Comment by hamishknight** [comment]
@swift-ci please smoke test

**Pull request merged**